src/implementation/Server/DaprServer.ts

```diff
@@ -50,43 +45,36 @@ export default class DaprServer {
   readonly actor: IServerActor;
   readonly client: DaprClient;
 
-  constructor(
-    serverHost?: string,
-    serverPort?: string,
-    daprHost?: string,
-    daprPort?: string,
-    communicationProtocol: CommunicationProtocolEnum = CommunicationProtocolEnum.HTTP,
-    clientOptions: Partial<DaprClientOptions> = {},
-    serverOptions: DaprServerOptions = {},
-  ) {
-    this.serverHost = serverHost ?? Settings.getDefaultHost();
-    this.serverPort = serverPort ?? Settings.getDefaultAppPort(communicationProtocol);
-    this.serverOptions = serverOptions;
-    this.daprHost = daprHost ?? Settings.getDefaultHost();
-    this.daprPort = daprPort ?? Settings.getDefaultPort(communicationProtocol);
-
-    clientOptions.daprHost = this.daprHost;
-    clientOptions.daprPort = this.daprPort;
-    clientOptions.communicationProtocol = communicationProtocol;
-
+  constructor(serverOptions: Partial<DaprServerOptions> = {}) {
+    const communicationProtocol = serverOptions.communicationProtocol ?? Settings.getDefaultCommunicationProtocol();
+    const clientOptions = getClientOptions(serverOptions.clientOptions, communicationProtocol);
```

Comment:
Should we do this? Client and Server cannot have different protocols this way, but it should be flexible since our future plans are to separate client and server packages. WDYT?

/cc @XavierGeerinck

Reply:
No, actually, preference is in this order: 
1. If Client options provide communication protocol, that is used.
2. Else if, Server options provide communication protocol, that is used
3. else, HTTP is used.

If someone wants to use different communication protocol, they can provide differet communication protocol in client options. I have put 2nd options so that, if someone wants to use same communication protocol, then they don't need to provide at two places mandatorily.

Reply:
I see. I guess this is the best UX then. Thanks for clarifying!

Reply:
Can we document this in DaprServer initialization docs?

Reply:
So, now a user can give, either:
```
const server = new DaprServer({
  serverHost,
  serverPort,
  communicationProtocol: CommunicationProtocolEnum.GRPC,
  clientOptions: {
    daprHost,
    daprPort,
  },
});
```
Or, they can also give:
```
const server = new DaprServer({
  serverHost,
  serverPort,
  communicationProtocol: CommunicationProtocolEnum.GRPC,
  clientOptions: {
    daprHost,
    daprPort,
    communicationProtocol: CommunicationProtocolEnum.HTTP,
  },
});
```

If we don't keep 2nd option, then they would have to give:
```
const server = new DaprServer({
  serverHost,
  serverPort,
  communicationProtocol: CommunicationProtocolEnum.GRPC,
  clientOptions: {
    daprHost,
    daprPort,
    communicationProtocol: CommunicationProtocolEnum.GRPC, (this becomes mandatory)
  },
});
```
or 
```
const server = new DaprServer({
  serverHost,
  serverPort,
  communicationProtocol: CommunicationProtocolEnum.GRPC,
  clientOptions: {
    daprHost,
    daprPort,
    communicationProtocol: CommunicationProtocolEnum.HTTP,
  },
});
```

Reply:
> Can we document this in DaprServer initialization docs?

Yes, sure. It makes sense to add it in Docs.